git_workflow_quick_reference.md

```diff
@@ -0,0 +1,159 @@
+```
+git checkout develop
+git pull origin develop
+```
+
+Switch back to your feature branch and update it with the changes from the main branch. Intrepid generally uses a "rebase" workflow for keeping feature branches up-to-date with the main branch. It's very important to understand exactly what rebase does, and so we recommend reading more *here*. (TODO: Add short description of rebase.) 
+
+```
+git checkout phil/JIRA-1234/profile
+git rebase -i develop
+```
+Note: The rebase `-i` flag is for "interactive" rebase, where you can choose which of your commits get applied, change their commit messages, and much more. Even if you do not need any of these advanced features, we still advise using `-i` just so that you can confirm that the right set of commits are being applied.
+
+At this point you may need to resolve conflicts.
+
+### Resolving conflicts
+
+First determine which files have conflicts.
+
+```
+git status
+```
+
+Conflicted files will be marked as "unmerged paths". (TODO: verify language.) To resolve the conflicts, open these files in your editor of choice and find where git has marked the conflict, which will look something like:
+```
+HEAD >>>>>>>>>>>>>>>>>
+myVariable = someValue;
+=====================
+myVariable = someOtherValue;
+[some commit] <<<<<<<<<<<<<<<<<<<
+```
+TODO: Get a more accurate example.
+
+Manually fix the conflict by removing git's annotations, and making sure that the combined code is correct. Then mark the file as unconflicted by staging it to be committed:
+```
+git add SomeFile.java
+```
+
+After resolving conflicts, you should generally verify that the code runs as expected. Note that during a rebase, not all commits may have been applied yet, and so you should check for behavior that matches the current commit. When you are satisfied with the changes:
+
+TODO: Talk more about fixing conflicts for an individual commit, and the various gotchas.
+
+```
+git rebase --continue
+```
+
+This should open an editor, to allow you to edit the commit message. Save and exit to continue. At this point the rebase will begin applying the next commit, which may cause more conflicts. Repeat until all commits have been applied, at which point git should show you the message "Rebase done." (TODO: Verify language.)
+
+### Push your changes to the remote
+
+Pushing your changes lets other developers access them, and also backs them up in the remote repository. Pushing regularly is good practice, as it keeps others in the loop and prevents you from losing work. To push your branch:
+
+```
+git push origin phil/JIRA-1234/profile-page
+```
+
+Note: If this command fails with a message "(TODO: Fill in this message.)" then that means that your local branch has diverged from a version previously pushed to the server. This may happen after a rebase. Read *this* for more on how to work around this issue.
+
+### Post your changes for review
+
+Once your branch is pushed, you should notifiy other developers to review it.
```

Comment:
Alternatively, we can look into other review systems than Github's. The one we use at Onshape can send notifications to specified reviewers when a review is updated. It also has some other nice features like marking issues as fixed or dropped, compare the diffs between review updates, and batch the review comment notifications together so that you don't get an email for each comment.